tests/unittests/memory_manager_tests.cpp

```diff
@@ -0,0 +1,145 @@
+#include <boost/test/unit_test.hpp>
+
+#include "memory_manager.h"
+
+BOOST_AUTO_TEST_SUITE(memory_manager_tests)
```

Comment:
If we are going to introduce unit testing framework can we split it into separate PR for review

Reply:
actually, this will not break any existing things. and the changes under unittests folder is to add the unit testing framework. It is not a very complicated PR. I do not think we need to split it into a separate PR.

Reply:
to speed up the review process, I have created another PR which only adds the unit test project. https://github.com/microsoft/DiskANN/pull/365